aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -276,11 +269,9 @@ void _validate_compressed_sparse_indices_kernel(
         at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
 
     const auto idx_ndims = idx.dim();
-    std::array<int64_t, MAX_DIMS> idx_sizes, idx_strides;
-    for (int i = 0; i < idx_ndims; i++) {
-      idx_sizes[i] = idx.size(i);
-      idx_strides[i] = idx.stride(i);
-    }
+
+    Tensor idx_sizes_ = at::tensor(idx.sizes(), idx.options().dtype(kLong));
+    Tensor idx_strides_ = at::tensor(idx.strides(), idx.options().dtype(kLong));
```

Comment:
minor nit: we can create a buffer tensor on the cpu, copy sizes and strides into it, move it to CUDA, then `unbind` it into two separate CUDA tensors for only a single data transfer. But maybe it is an over-optimization...

Reply:
Done.